server/routes/users.js

```diff
@@ -0,0 +1,78 @@
+const router = require('express').Router();
+const bcrypt = require('bcrypt')
+const { User, Session } = require('../db');
+const Events = require('../db/models/Events');
+
+router.get('/', async (req, res, next) => {
+  try {
+    const users = await User.findAll();
+    res.send(users);
+  } catch (err) {
+    next(err);
+  }
+});
+router.get('/get-user', (req, res, next) => {
+  try {
+      res.send(req.user)
+  }
+  catch (err) {
+      next(err)
+  }
+})
+
+router.get('/:id', async (req, res, next) => {
+  try {
+    const singleUser = await User.findByPk(req.params.id, { include: [Events] });
+    if (!singleUser) {
+      const error = new Error('USER NOT FOUND');
+      error.status = 404;
+      throw error;
+    }
+    res.status(200).send(singleUser);
+  } catch (err) {
+    next(err);
+  }
+});
+
+router.delete('/:id', async (req, res, next) => {
+  try {
+    const singleUser = await User.findByPk(req.params.id);
+    await singleUser.destroy();
+    res.sendStatus(204);
+  } catch (err) {
+    next(err);
+  }
+});
+
+
+router.put('/follow', async (req, res, next) => {
+  try {
+    const followedUser = await User.findByPk(req.body.id);
+    const user=await followedUser.update({friends: req.body.arr });    
```

Comment:
Users.hasMany(Users, { as: 'follower' });

--in api--
user.addFriend(otherUser)